x-pack/metricbeat/module/aws/utils_test.go

```diff
@@ -434,3 +434,54 @@ func TestGetResourcesTags(t *testing.T) {
 	}
 	assert.Equal(t, expectedResourceTagMap, resourceTagMap)
 }
+
+func TestGetStartTimeEndTime(t *testing.T) {
+	var cases = []struct {
+		title         string
+		start         string
+		period        time.Duration
+		latency       time.Duration
+		expectedStart string
+		expectedEnd   string
+	}{
+		// window should align with period e.g. requesting a 5 minute period at 10:27 gives 10:20->10:25
+		{"1 minute", "2022-08-15T13:38:45Z", time.Second * 60, 0, "2022-08-15T13:37:00Z", "2022-08-15T13:38:00Z"},
+		{"2 minutes", "2022-08-15T13:38:45Z", time.Second * 60 * 2, 0, "2022-08-15T13:36:00Z", "2022-08-15T13:38:00Z"},
+		{"3 minutes", "2022-08-15T13:38:45Z", time.Second * 60 * 3, 0, "2022-08-15T13:33:00Z", "2022-08-15T13:36:00Z"},
```

Comment:
just curious, why the start time and end time is not "2022-08-15T13:35:00Z", "2022-08-15T13:38:00Z"?